### PR TITLE
Fix Data Masking

### PR DESCRIPTION
**Pull request by xTEddie**
### Description
- Data Masking is enabled while the flag was set to false but with active data masking rules

## Solution Proposed
- Added a null/undefined check

### Acceptance criteria
- Data masking should be working as expected

## Test cases and evidence
- Validate data masking is working accordingly

### Sanity Tests
-   [ ] You have tested all changes in Popout mode
-   [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
-   [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y
- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__

### CODE REVIEW

**Comment by kevinlin98z** [comment]
Please add change log

**Review by kevinlin98z — APPROVED** [review]

File: chat-widget/src/components/webchatcontainerstateful/common/utils/isMaskingFromCustomer.ts [hunk 1]
```diff
@@ -9,8 +10,8 @@ export const isMaskingforCustomer = (maskingInfo: IDataMaskingInfo): boolean =>
 
     // If the masking rules are provided and;
     // If the masking setting is NOT null and masking for customer is NOT null, return the configuration
-    if (maskingInfo.setting?.msdyn_maskforcustomer) {
-        return maskingInfo.setting.msdyn_maskforcustomer;
+    if (!isNullOrUndefined(maskingInfo.setting?.msdyn_maskforcustomer)) {
+        return parseLowerCaseString(maskingInfo?.setting?.msdyn_maskforcustomer) === "true";
     }
```
Comment (kevinlin98z) [inline chat-widget/src/components/webchatcontainerstateful/common/utils/isMaskingFromCustomer.ts:15]: Please add to change log

File: chat-widget/src/common/utils.ts [hunk 2]
```diff
@@ -438,4 +438,8 @@ export const  formatTemplateString = (templateMessage : string, values : any) =>
     return templateMessage.replace(/{(\d+)}/g, (match, index) => {
         return typeof values[index] !== "undefined" ? values[index] : match;
     });
+};
+
+export const parseLowerCaseString = (property: string | boolean | undefined): string => {
```
Comment (elopezanaya) [inline chat-widget/src/common/utils.ts:443]: add tests to https://github.com/microsoft/omnichannel-chat-widget/blob/main/chat-widget/src/common/utils.test.ts

**Review by elopezanaya — CHANGES_REQUESTED** [review]
add unit tests to validate the functionality

**Review by elopezanaya — APPROVED** [review]

**Pull request merged**